phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewTTLIT.java

```diff
@@ -0,0 +1,111 @@
+            int expectedRows) throws IOException, SQLException {
+
+        FilterList filterList = new FilterList(FilterList.Operator.MUST_PASS_ALL);
+        RowFilter schemaNameFilter = new RowFilter(
+                CompareFilter.CompareOp.EQUAL,
+                new SubstringComparator(schemaName)
+        );
+        ColumnPrefixFilter viewTTLPrefixFilter = new ColumnPrefixFilter(PhoenixDatabaseMetaData.VIEW_TTL_BYTES);
+        filterList.addFilter(schemaNameFilter);
+        filterList.addFilter(viewTTLPrefixFilter);
+        try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+                .getTable(PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME_BYTES)) {
+
+            Scan allRows = new Scan();
+            allRows.setRaw(true);
+            allRows.setTimeRange(minTimestamp, HConstants.LATEST_TIMESTAMP);
+            allRows.setFilter(filterList);
+            ResultScanner scanner = tbl.getScanner(allRows);
+            int numMatchingRows = 0;
+            for (Result result = scanner.next(); result != null; result = scanner.next()) {
+                numMatchingRows +=
+                        result.containsColumn(QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES,
+                                PhoenixDatabaseMetaData.VIEW_TTL_BYTES) ? 1 : 0;
+            }
+            assertEquals(String.format("Expected rows do match for table = %s at timestamp %d",
+                    Bytes.toString(PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME_BYTES), minTimestamp), expectedRows, numMatchingRows);
+        }
+
+    }
+
+    /**
+         * -----------------
+         * Test methods
+         * -----------------
+         */
+
+    @Test
+    public void testWithBasicTenantViewWithNoViewTTLDefined() throws Exception {
+
+        long startTime = System.currentTimeMillis();
+        // Define the test schema.
+        PhoenixTestBuilder.SchemaBuilder.TableOptions
+                tableOptions = PhoenixTestBuilder.SchemaBuilder.TableOptions.withDefaults();
+        tableOptions.getTablePKColumns().add("ID");
+        tableOptions.getTablePKColumnTypes().add("CHAR(15)");
+        tableOptions.getTableColumns().clear();
+        tableOptions.getTableColumnTypes().clear();
+
+        final PhoenixTestBuilder.SchemaBuilder schemaBuilder = new PhoenixTestBuilder.SchemaBuilder(getUrl());
+        schemaBuilder.withTableOptions(tableOptions).build();
+
+        // Define the test data.
+        PhoenixTestBuilder.DataSupplier dataSupplier = new PhoenixTestBuilder.DataSupplier() {
+
+            final String
+                    orgId =
+                    String.format("00D0x000%s", schemaBuilder.getDataOptions().getUniqueName());
+            final String kp = SchemaUtil.normalizeIdentifier(schemaBuilder.getEntityKeyPrefix());
+
+            @Override public List<Object> getValues(int rowIndex) {
+                String id = String.format("00A0y000%07d", rowIndex);
```

Comment:
What is this logic for getting the `id`?

Reply:
Nothing special! Just trying to get char(15) id, just using the sfdc id format.

Reply:
Can you add a comment about the format of the generated key here?

Reply:
Maybe move this 15 char format to `Util` and named some method like `getRowIndexFormat` so that we can reuse it and have better code readability?

Reply:
For the time being, created a static for it.